fieldMapping.json

```diff
@@ -0,0 +1,209 @@
+          "Value": "\"Person.ExternalId\"",
+          "UsedInNotifications": false,
+          "StoreInAccountData": true
+        }
+      ]
+    },
+    {
+      "Name": "id",
+      "Description": "",
+      "Type": "Text",
+      "MappingActions": [
+        {
+          "MapForActions": [
+            "Create",
+            "Update"
+          ],
+          "MappingMode": "None",
+          "Value": "null",
+          "UsedInNotifications": false,
+          "StoreInAccountData": true
+        }
+      ]
+    },
+    {
+      "Name": "email",
+      "Description": "",
+      "Type": "Text",
+      "MappingActions": [
+        {
+          "MapForActions": [
+            "Create",
+            "Update"
+          ],
+          "MappingMode": "Complex",
+          "Value": "\"function GetEmailAddress(){\\r\\n    return Person.Accounts._bccd667807a9431a9d4acfc64c19a3d6.mail\\r\\n}\\r\\nGetEmailAddress()\"",
+          "UsedInNotifications": false,
+          "StoreInAccountData": false
+        }
+      ]
+    },
+    {
+      "Name": "supervisorid",
+      "Type": "Text",
+      "MappingActions": [
+        {
+          "MapForActions": [
+            "Create",
+            "Update"
+          ],
+          "MappingMode": "Field",
+          "Value": "\"Person.PrimaryManager.ExternalId\"",
+          "UsedInNotifications": false,
+          "StoreInAccountData": false
+        }
+      ]
+    },
+    {
+      "Name": "startdate",
+      "Description": "",
+      "Type": "Text",
+      "MappingActions": [
+        {
+          "MapForActions": [
+            "Create",
+            "Update"
+          ],
+          "MappingMode": "Complex",
+          "Value": "\"const formatedTimestamp = () => {\\r\\n    const startDate = Person.PrimaryContract.StartDate\\r\\n    const date = startDate.toISOString().split('T')[0];\\r\\n    const time = \\\"00:00:00\\\" //startDate.toTimeString().split(' ')[0].replace(/:/g, ':');\\r\\n    return `${date}T${time}Z`\\r\\n}\\r\\n\\r\\nfunction GetFormattedStartDate(){\\r\\n    let formattedDate = formatedTimestamp();\\r\\n    return formattedDate\\r\\n}\\r\\nGetFormattedStartDate()\"",
+          "UsedInNotifications": false,
+          "StoreInAccountData": false
+        }
+      ]
+    },
+    {
+      "Name": "birthdate",
+      "Description": "",
+      "Type": "Text",
+      "MappingActions": [
+        {
+          "MapForActions": [
+            "Create",
+            "Update"
+          ],
+          "MappingMode": "Complex",
+          "Value": "\"const formatedBirthDatestamp = () => {\\r\\n    const birthDate = Person.Details.BirthDate\\r\\n    const date = birthDate.toISOString().split('T')[0];\\r\\n    const time = \\\"00:00:00\\\" //birthDate.toTimeString().split(' ')[0].replace(/:/g, ':');\\r\\n    return `${date}T${time}Z`\\r\\n}\\r\\n\\r\\nfunction GetFormattedBirthDate(){\\r\\n    let formattedDate = formatedBirthDatestamp();\\r\\n    return formattedDate\\r\\n}\\r\\nGetFormattedBirthDate()\"",
```

Comment:
Did you check what happens if the BirthDate is empty? Maybe a the following code needs to be added to make sure the javescript won't fail.

```JavaScript
if (typeof Person.Details.BirthDate !== 'undefined' && Person.Details.BirthDate) {

}
```

Reply:
Added check on existance.